src/main/java/sirius/kernel/cache/SizedCache.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.kernel.cache;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import sirius.kernel.Sirius;
+import sirius.kernel.health.Exceptions;
+
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+/**
+ * Defines a size limited mapping from key to values.
+ * <p>
+ * Items can be put in the cache by direct method or calculated from a value computer, either a default
+ * one provided when initializing the class and/or a custom one when getting an item.
+ * <p>
+ * The maximum size of the cache is given during initialization, either with a fixed value or by providing
+ * a configuration key where the value should be retrieved from.
+ * <p>
+ * Note that a {@link ManagedCache} is the preferred solution for most cases, but certain situations might
+ * require a pure local cache, such as a temporary big data load where cache polution is not desired.
+ * In such cases, this class will give the comfort and ease of use of a Map-like implementation eliminating
+ * the risk of an out-of-memory.
+ *
+ * @param <T> the type of object to be stored in the cache
+ */
+public class SizedCache<T> {
+
+    private final LoadingCache<String, T> cache;
+    private final boolean hasDefaultLoader;
+
+    /**
+     * Creates a sized cache with a maximum size and default value computer.
+     *
+     * @param maximumSize          the maximum size of the cache
+     * @param defaultValueComputer the default function called to compute the value of a missing entry
+     */
+    public SizedCache(long maximumSize, Function<String, T> defaultValueComputer) {
+        if (maximumSize <= 0) {
+            throw new IllegalArgumentException("The cache maximum size must be higher than zero.");
+        }
+        this.cache = CacheBuilder.newBuilder().maximumSize(maximumSize).build(new CacheLoader<String, T>() {
+            @Override
+            public T load(String key) throws Exception {
+                if (defaultValueComputer == null) {
+                    return null;
+                }
+                return defaultValueComputer.apply(key);
+            }
+        });
+        this.hasDefaultLoader = defaultValueComputer != null;
+    }
+
+    /**
+     * Creates a sized cache with a maximum size and default value computer.
+     *
+     * @param configKey            the name of a configuration key containing the maximum size of the cache
+     * @param defaultValueComputer the default function called to compute the value of a missing entry
+     */
+    public SizedCache(String configKey, Function<String, T> defaultValueComputer) {
```

Comment:
I'd remove this as provide a config value leaves the false impression that we could control the cache size - we cannot, as there can be many instances of the same cache.